src/core/restaurants/getRandomRestaurantSheets.tsx

```diff
@@ -0,0 +1,47 @@
+"use server";
+
+import { Restaurant } from "@prisma/client";
+import prisma from "../../../prisma/singleton";
+import { RestaurantType } from "./restaurantType";
+import { Diet } from "./diet";
+import { getAllRestaurantsSheets } from "./getAllRestaurantsSheets";
+import { RestaurantInfos } from "./interface";
+
+export const getRandomRestaurantSheets = async (
+  restaurantType: RestaurantType,
+  diet: Diet
+): Promise<RestaurantInfos> => {
+  const restaurants = await getAllRestaurantsSheets();
+
+  const whereClause = {
+    canEatIn: restaurantType === RestaurantType.EAT_IN ? true : undefined,
+    canTakeAway: restaurantType === RestaurantType.TAKE_AWAY ? true : undefined,
+    vegetarianFriendly: diet === Diet.VEGETARIAN ? true : undefined,
+    meatLover: diet === Diet.MEATLOVER ? true : undefined,
+  };
+  console.log("whereClause : ", whereClause)
+  
+  type CorrectKey = keyof RestaurantInfos
+
+
+  // Correct clause keys are the keys of the RestaurantInfos interface
+  function isCorrectKey(key: any, restaurant: RestaurantInfos): key is CorrectKey {
+    const expectedKeys = Object.keys(restaurant)
+    return expectedKeys.includes(key)
+  }
```

Comment:
```suggestion
  function isRestaurantInfoKey(key: any, restaurant: RestaurantInfos): key is CorrectKey {
    const expectedKeys = Object.keys(restaurant)
    return expectedKeys.includes(key)
  }
```

Naming > Comments ;)

Reply:
Okay